_pytest/fixtures.py

```diff
@@ -1130,7 +1130,40 @@ def getfixturedefs(self, argname, nodeid):
         else:
             return tuple(self._matchfactories(fixturedefs, nodeid))
 
+    @classmethod
+    def _splitnode(cls, nodeid):
+        """Split a nodeid into constituent 'parts'.
+
+        Node IDs are strings, and can be things like:
+            '',
+            'testing/code',
+            'testing/code/test_excinfo.py'
+            'testing/code/test_excinfo.py::TestFormattedExcinfo::()'
+
+        """
+        if nodeid == '':
+            return []
+        sep = py.path.local.sep
+        parts = nodeid.split(sep)
+        last_part = parts[-1]
+        if '::' in last_part:
+            # Replace single last element 'test_foo.py::Bar::()' with multiple elements 'test_foo.py', 'Bar', '()'
+            parts[-1:] = last_part.split("::")
+        return parts
```

Comment:
Just a suggestion to simplify the code: remove the special case for the empty `nodeid`. The following would return functionally equivalent results (it would just return `['']` instead of `[]`) without special-casing anything:
```
sep = py.path.local.sep
parts = nodeid.split(sep)
parts[-1:] = parts[-1].split("::")
return parts
```

Reply:
Thanks, I've shortened this down a bit (and made the docstring longer to compensate ;-))